ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/Follower.hs

```diff
@@ -516,67 +519,30 @@ forward registry varFollower blockComponent CDB{..} = \pts -> do
           FollowerInit                  -> return ()
           FollowerInMem _               -> return ()
 
--- | Update the given 'FollowerState' to account for switching the current
--- chain to the given fork (which might just be an extension of the
--- current chain).
---
--- PRECONDITION: the intersection point must be within the fragment bounds
--- of the new chain
+-- | Switches the follower to the new fork, by checking whether the follower is
+-- following an old fork, and updating the follower state to rollback to the
+-- intersection point if it is.
 switchFork ::
-     forall m blk b. (HasHeader blk, HasHeader (Header blk))
-  => Point blk  -- ^ Intersection point between old and new chain
-  -> AnchoredFragment (Header blk)  -- ^ The new chain
-  -> FollowerState m blk b -> FollowerState m blk b
-switchFork ipoint newChain followerState =
-    assert (AF.withinFragmentBounds (castPoint ipoint) newChain) $
-      case followerState of
-        -- If the follower is still in the initial state, switching to a fork
-        -- won't affect it.
-        FollowerInit             -> followerState
-        -- If the follower is still reading from the ImmutableDB, switching to a
-        -- fork won't affect it.
-        FollowerInImmutableDB {} -> followerState
-        FollowerInMem rollState  ->
-            case pointSlot followerPoint `compare` pointSlot ipoint of
-              -- If the follower point is more recent than the intersection point,
-              -- we have to roll back the follower to the intersection point.
-              GT -> FollowerInMem $ RollBackTo ipoint
```

Comment:
Isn't it true that combined with the invariant that the follower is always on the current chain this implies that `followerPoint` is part of `oldPoints`?

Reply:
The situation I was trying to refer to is one in which `followerPoint` is in the new chain. But I guess that is not possible right, because there should be no followers in the new chain.

Reply:
It seems like this situation has the follower point after the intersection point:


     G -- b1 -- b2
                  \- b3

when at the time of adding b3, the follower is at b2.

Reply:
> It seems like this situation has the follower point after the intersection point:


But in this case `b2` would be in `oldChain - newChain` right? So we're covering this case, aren't we?

Reply:
Yes